clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -234,6 +234,18 @@ public long append(long offset, long timestamp, byte[] key, byte[] value) {
     }
 
     /**
+     * Append a new record at the next consecutive offset. If no records have been appended yet, use the base
+     * offset of this builder.
+     * @param timestamp The record timestamp
+     * @param key The record key
+     * @param value The record value
+     * @return crc of the record
+     */
+    public long append(long timestamp, byte[] key, byte[] value) {
```

Comment:
Hmm, can we think of another name for the methods that increment the offset automatically?